app/controllers/application_controller.rb

```diff
@@ -14,6 +14,12 @@ class ApplicationController < ActionController::Base
     end
   end
 
+  def authenticate_admin_user!
+    unless current_user && current_user.admin?
```

Comment:
Nit. What do you think about using the safe navigation operator here to keep it terse?

```ruby
unless current_user&.admin?
```

Reply:
To be completely honest I think it's the best idea since your hair Arun.

Reply:
I don't think we should use `&.` at all. The syntax is awkward and imo harder to read. It also encourages bad practices with violating law of Demeter.

Reply:
I'm personally with Arun here. That syntax was created specifically because of the kind of problem where you need to check for an attribute but need to avoid the nil method error.

What is the violation to Demeter?

Reply:
> need to avoid the nil method error

That usually signifies theres a deeper problem somewhere. There are better ways of getting around those kind of situations. I think if we start introducing safe navigation, we will be shooting ourselves in the foot in the long run by making the code base increasingly fragile and error prone.

Have a read over this [article](https://karolgalanciak.com/blog/2017/09/24/do-or-do-not-there-is-no-try-object-number-try-considered-harmful/) if you get a chance. It's about the problems with `try` including violation of law of Demeter , but they all apply to the safe navigation as well.